masonry_core/src/core/action.rs

```diff
@@ -2,18 +2,23 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use std::any::Any;
+use std::fmt::Debug;
 
-use crate::core::PointerButton;
+use crate::core::{PointerButton, WidgetId};
 
 // TODO - Replace actions with an associated type on the Widget trait
 // See https://github.com/linebender/xilem/issues/664
 
+// TODO - Document
+#[derive(Debug)]
+pub struct Action(pub Box<dyn DebugAndAny + Send>, pub WidgetId);
+
 // TODO - TextCursor changed, ImeChanged, EnterKey, MouseEnter
 #[non_exhaustive]
 /// Events from UI elements.
 ///
 /// Note: Actions are still a WIP feature.
-pub enum Action {
+pub enum DefaultAction {
```

Comment:
Can we document that this is an intermediate state. I think for reviewing, it wouldn't be unreasonable to leave this as-is, and then split it out as a follow-up